### DISCUSSION TITLE
How to ensure all DNS requests on local network redirect using dnscrypt-proxy

### DISCUSSION THREAD

**Discussion by jaYINGLING** [DNSCrypt/dnscrypt-proxy#2366] [Q&A]
Hello, I am using edgeOS for a family home network and am attempting to do content filtering by using dnscrypt-proxy to connect via DoH to the cloudflare family DNS servers. The solution works well as long as any DNS requests are made to the default nameserver (192.168.1.1), but the filtering is easily circumvented by a user specifying their own DNS server. 

I understand that this can be solved using NAT to redirect port 53 requests to 192.168.1.1, but haven't been able to configure it correctly. 

Anyone have guidance on how to get this working? Here is my destination NAT rule for port 53.
![image](https://user-images.githubusercontent.com/37126096/230655329-33df54ad-aeac-40f3-ba6e-65beaef1f654.png)

**Comment by Menard01** [comment]
Though my network is only me, desktop PC with ethernet connection to my internet routeur, I have the same question, but I can give an idea, some tutorial on the internet about dnscrypt-proxy talk about some tool in it called "hostip" but I see nothing about this in the official dnscrypt-proxy documentation, this tool is supposed to answer our problem 
I personnaly ask about this for some applications that could bypass the proxy

**Comment by jaYINGLING** [comment]
Can you share the link where you read about this?

**Comment by Menard01** [comment]
There https://doc.ubuntu-fr.org/dnscrypt but now I have doubts because it seems to be only an equivalent of "dig" from the bind9-dnsutils package
Perhaps we can try also dnstop tool

**Comment by bmeirellesRJ** [comment]
i have a similar problem. I'm using these rule in my firewall (nftables, iptables substitute) to capture dns traffic and redirect to my localhost dns server:
nft add rule inet nat PREROUTING iifname "br0" meta l4proto {tcp, udp} th dport 53 redirect to :53

In iptables the equivalent is:
iptables -t nat -A PREROUTING -s 192.168.1.0/24 -p udp --dport 53 -j REDIRECT --to-port 53
ip6tables -t nat -A PREROUTING -s fd00:ffff:fffd:100::/64  -p udp --dport 53 -j REDIRECT --to-port 53

With ipv4 it works perfectly, but with ipv6 it doesn't repply.

The problem is with dnscrypt-proxy, because with bind9 it works perfectly ipv4 and ipv6.

My version is 2.1.4 and my socket configuration is listen_addresses =  ['[::]:53']
